src/qat/purr/compiler/devices.py

```diff
@@ -315,6 +315,7 @@ def __init__(
         acquire_allowed: bool = False,
         pulse_channel_min_frequency: float = 0.0,
         pulse_channel_max_frequency: float = np.inf,
+        grid_time: float = 4e-09,  # 4 ns
```

Comment:
ah, it's seconds, I'd be very careful with arithmetic operations on non integer types. Here it's fine, keep it as seconds, when you use the field, make the necessary conversions

Reply:
in general this verification feels like it should be a constant rather than an instance field. Ideally you'd have some enum `Constants` and use it like `Constants.GRID_TIME`.

I also think it should not belong here and should rather be implemented internally as it's backend-specific.

Reply:
Moved internally.